conu/backend/podman/container.py

```diff
@@ -0,0 +1,330 @@
+            return metadata["Config"].get("Image", None)
+        return None
+
+    def wait_for_port(self, port, timeout=10, **probe_kwargs):
+        """
+        block until specified port starts accepting connections, raises an exc ProbeTimeout
+        if timeout is reached
+
+        :param port: int, port number
+        :param timeout: int or float (seconds), time to wait for establishing the connection
+        :param probe_kwargs: arguments passed to Probe constructor
+        :return: None
+        """
+        Probe(timeout=timeout, fnc=functools.partial(self.is_port_open, port), **probe_kwargs).run()
+
+    def delete(self, force=False, **kwargs):
+        """
+        remove this container; kwargs indicate that some container runtimes
+        might accept more parameters
+
+        :param force: bool, if container engine supports this, force the functionality
+        :return: None
+        """
+        cmdline = ["podman", "rm", "--force" if force else "", self.get_name()]
+        run_cmd(cmdline)
+
+    def mount(self, mount_point=None):
+        """
+        mount container filesystem
+
+        :return: str, the location of the mounted file system
+        """
+        cmd = ["podman", "mount", self._id or self.get_id()]
+        output = run_cmd(cmd, return_output=True).rstrip("\n\r")
+        return output
+
+    def umount(self, all=False, force=True):
+        """
+        unmount container filesystem
+        :param all: bool, option to unmount all mounted containers
+        :param force: bool, force the unmounting of specified containers' root file system
+        :return: str, the output from cmd
+        """
+        # FIXME: handle error if unmount didn't work
+        options = []
+        if force: options.append('--force')
+        if all: options.append('--all')
+        cmd = ["podman", "umount"] + options + [self.get_id() if not all else ""]
+        return run_cmd(cmd, return_output=True)
+
+    def logs(self, follow=False):
+        """
+        Get logs from this container. Iterator has one log line followed by a newline in next item.
+        The logs are NOT encoded (they are str, not bytes).
+
+        Let's look at an example::
+
+            image = conu.PodmanImage("fedora", tag="27")
+            command = ["bash", "-c", "for x in `seq 1 5`; do echo $x; sleep 1; done"]
+            container = image.run_via_binary(command=command)
+            for line in container.logs(follow=True):
+                print(line)
+
+        This will output
+
+        .. code-block:: none
+
+            '1' '\n' '2' '\n' '3' '\n' '4' '\n' '5' '\n'
+
+        :param follow: bool, provide new logs as they come
+        :return: iterator (of str)
+        """
+        # TODO: podman logs have different behavior than docker
+        follow = ["--follow"] if follow else []
+        cmdline = ["podman", "logs"] + follow + [self._id or self.get_id()]
+        output = run_cmd(cmdline, return_output=True)
```

Comment:
I wonder if we can tinker run_cmd make this output a generator.

Reply:
as we discussed in person, this can be done in a subsequent PR (feel free to put to comments as TODO)